address_book/contacts/forms.py

```diff
@@ -0,0 +1,26 @@
+from django import forms
+from django.core.validators import RegexValidator
+from contacts.models import Contact
+
+class ContactForm(forms.ModelForm):
+    phone_number = forms.CharField(max_length=15, validators=[
+        RegexValidator(r'^\d{1,15}$', message="Phone number must be numeric.")
+    ])
+
+    class Meta:
+        model = Contact
+        fields = ['first_name', 'last_name', 'email', 'phone_number', 'address']
+
+    def clean_phone_number(self):
+        phone_number = self.cleaned_data['phone_number']
+        try:
+            # Check if any contact with this phone number exists
+            contacts = Contact.objects.filter(phone_number=phone_number)
```

Comment:
As pointed out in the last PR, please use get_object_or_404 here or add .first() for this filtering.